flink-runtime/src/main/java/org/apache/flink/runtime/messages/webmonitor/JobStatusesWithIdsOverview.java

```diff
@@ -0,0 +1,235 @@
+import org.apache.flink.runtime.rest.messages.json.JobIDSerializer;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * An overview of how many jobs are in which status.
+ */
+public class JobStatusesWithIdsOverview implements ResponseBody, InfoMessage {
+
+	private static final long serialVersionUID = -3699051943490133183L;
+
+	public static final String FIELD_NAME_JOBS_CREATED_IDS = "jobs-created";
+	public static final String FIELD_NAME_JOBS_RUNNING_IDS = "jobs-running";
+	public static final String FIELD_NAME_JOBS_FINISHED_IDS = "jobs-finished";
+	public static final String FIELD_NAME_JOBS_CANCELLING_IDS = "jobs-cancelling";
+	public static final String FIELD_NAME_JOBS_CANCELLED_IDS = "jobs-cancelled";
+	public static final String FIELD_NAME_JOBS_FAILING_IDS = "jobs-failing";
+	public static final String FIELD_NAME_JOBS_FAILED_IDS = "jobs-failed";
+	public static final String FIELD_NAME_JOBS_RESTARTING_IDS = "jobs-restarting";
+	public static final String FIELD_NAME_JOBS_SUSPENDED_IDS = "jobs-suspended";
+	public static final String FIELD_NAME_JOBS_RECONCILING_IDS = "jobs-reconciling";
+
+	@JsonProperty(FIELD_NAME_JOBS_CREATED_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsCreated;
+
+	@JsonProperty(FIELD_NAME_JOBS_RUNNING_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsRunningOrPending;
+
+	@JsonProperty(FIELD_NAME_JOBS_FINISHED_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsFinished;
+
+	@JsonProperty(FIELD_NAME_JOBS_CANCELLING_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsCancelling;
+
+	@JsonProperty(FIELD_NAME_JOBS_CANCELLED_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsCancelled;
+
+	@JsonProperty(FIELD_NAME_JOBS_FAILING_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsFailing;
+
+	@JsonProperty(FIELD_NAME_JOBS_FAILED_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsFailed;
+
+	@JsonProperty(FIELD_NAME_JOBS_RESTARTING_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsRestarting;
+
+	@JsonProperty(FIELD_NAME_JOBS_SUSPENDED_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsSuspended;
+
+	@JsonProperty(FIELD_NAME_JOBS_RECONCILING_IDS)
+	@JsonSerialize(contentUsing = JobIDSerializer.class)
+	private final List<JobID> jobsReconciling;
```

Comment:
I would actually not group the jobs here. Rather we could return a collection of `JobIDs` with `JobStatus`. If you need to present a grouped overview, then the client has to calculate it. That would simplify this class and the handler considerably. What do you think?